autoload/ale/c.vim

```diff
@@ -89,3 +89,59 @@ function! ale#c#FindCompileCommands(buffer) abort
 
     return ''
 endfunction
+
+" Given a buffer number, find a compilation database (compile_commands.json)
+" and parse this into an argument string
+function! ale#c#FindCompileArgs(buffer) abort
+    let l:use_args = []
+
+    let l:compile_commands_path = ale#path#FindNearestFile(a:buffer, 'compile_commands.json')
+    if (!empty(l:compile_commands_path))
+        let l:bufname = fnamemodify(bufname(a:buffer), ':p')
+        let l:command = ''
+        let l:file = ''
+
+        for elem in readfile(l:compile_commands_path)
+            let l:split = split(elem, ':')
```

Comment:
Thanks for the feedback. Bit swamped at work currently but will try and find some time to get back on this.